spec/suites/layer/GeoJSONSpec.js

```diff
@@ -512,3 +512,364 @@ describe("L.LayerGroup#toGeoJSON", function () {
+			var layer = L.GeoJSON.geometryToLayer({
+				type: "Feature",
+				geometry: {
+					type: "MultiPoint",
+					coordinates: [
+						[1, 2, 3], [4, 5, 6]
+					]
+				}
+			}, {
+				coordsToLatLng: customCoordstoLatLng
+			});
+			expect(layer.getLayers().map(function (lyr) {
+				return lyr.getLatLng();
+			})).to.eql([
+				{lat: 3, lng: 2, alt: 4},
+				{lat: 6, lng: 5, alt: 7}
+			]);
+		});
+	});
+
+	describe("#coordsToLatLng", function () {
+		it("creates a LatLng object", function () {
+			var latLng = L.GeoJSON.coordsToLatLng([1, 2]);
+			var latLngWithAlt = L.GeoJSON.coordsToLatLng([3, 4, 5]);
+			expect(latLng).to.be.a(L.LatLng);
+			expect(latLngWithAlt).to.be.a(L.LatLng);
+		});
+
+		it("creates a LatLng object with given coordinates", function () {
+			var latLng = L.GeoJSON.coordsToLatLng([1, 2]);
+			var latLngWithAlt = L.GeoJSON.coordsToLatLng([3, 4, 5]);
+			expect(latLng.lat).to.be(2);
+			expect(latLng.lng).to.be(1);
+			expect(latLng.alt).not.to.be.ok();
+			expect(latLngWithAlt.lat).to.be(4);
+			expect(latLngWithAlt.lng).to.be(3);
+			expect(latLngWithAlt.alt).to.be(5);
+		});
+	});
+
+	describe("#coordsToLatLngs", function () {
+
+		function customCoordsToLatLng(coords) {
+			return L.latLng(coords[1] + 1, coords[0] + 1, coords[2] + 1);
+		}
+
+		it("creates a multidimensional array of LatLngs", function () {
+			var latLngs = L.GeoJSON.coordsToLatLngs([[1, 2], [3, 4], [5, 6]]);
+			expect(latLngs.length).to.be(3);
+			latLngs.forEach(function (latLng) {
+				expect(latLng).to.be.a(L.LatLng);
+			});
+		});
+
+		it("creates a multidimensional array of LatLngs (levelsDeep=1)", function () {
+			var latLngs = L.GeoJSON.coordsToLatLngs([
+				[[1, 2], [3, 4], [5, 6]],
+				[[5, 6], [7, 8], [9, 10]]
+			], 1);
+			expect(latLngs.length).to.be(2);
+			latLngs.forEach(function (arr) {
+				arr.forEach(function (latlng) {
+					expect(latlng).to.be.a(L.LatLng);
+				});
+			});
+		});
+
+		it("creates a multidimensional array of LatLngs with custom coordsToLatLng", function () {
+			var coords = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
+			var latLngs = L.GeoJSON.coordsToLatLngs(coords, 0, customCoordsToLatLng);
+			expect(latLngs).to.eql([
+				{lat: 3, lng: 2, alt: 4},
+				{lat: 6, lng: 5, alt: 7},
+				{lat: 9, lng: 8, alt: 10}
+			]);
+		});
+
+		it("creates a multidimensional array of LatLngs with custom coordsToLatLng (levelDeep=1)", function () {
+			var coords = [
+				[[1, 2, 3], [4, 5, 6]],
+				[[12, 13, 14], [15, 16, 17]]
+			];
+			var latLngs = L.GeoJSON.coordsToLatLngs(coords, 1, customCoordsToLatLng);
+			expect(latLngs).to.eql([
+				[
+					{lat: 3, lng: 2, alt: 4},
+					{lat: 6, lng: 5, alt: 7}
+				],
+				[
+					{lat: 14, lng: 13, alt: 15},
+					{lat: 17, lng: 16, alt: 18}
+				]
+			]);
+		});
+	});
+
+	describe("#latLngToCoords", function () {
+		it("returns an array of coordinates and altitude", function () {
+			var coords = L.GeoJSON.latLngToCoords(L.latLng(2, 1));
```

Comment:
Here and in many places below: better replace occurrences `L.latLng(2, 1)` with arrays `[2,1]`.

Reply:
The docs for this function tell that a `L.LatLng` is used a input: https://leafletjs.com/reference-1.7.1.html#geojson-latlngtocoords.

Reply:
Sure, that is in specs, and your tests are correct. But it is also documented that you can pass plain array in every place where LatLng is expected.
Definitely you can double the number of tests and check both cases, but in my opinion that would be very excessive.
So I suggest you to keep just array form, because:
1) It is shorter and more expressive
2) You already prefer array in all the previous cases, so it make sense to keep single style.

Reply:
I can change that, but like i said in this case i tried to stick to the documentation. The other cases like `coordsToLatLng` or `coordsToLatLng` accept, according to the docs, an array of coordinates and thats the reason i chose the array over `LatLng`.

Reply:
As I said your tests are correct but too verbose (as for my taste).

And to be consistent you also should consider [this](https://leafletjs.com/reference-1.7.1.html#latlng) part of docs:
> All Leaflet methods that accept LatLng objects also accept them in a simple Array form and simple object form (unless noted otherwise).

P.S.
I've also find these 2 cases where you are not checking resulting types:

https://github.com/Leaflet/Leaflet/blob/cc74c2c8a11dcc4574afa1d746cb2dc0696a31ea/spec/suites/layer/GeoJSONSpec.js#L729

https://github.com/Leaflet/Leaflet/blob/cc74c2c8a11dcc4574afa1d746cb2dc0696a31ea/spec/suites/layer/GeoJSONSpec.js#L739

It's not so important (for my taste), because such type of error hard to make by accident, and it will manifest itself in some other part anyway.
But it also makes sense to keep all your tests consistent, so...

Reply:
> All Leaflet methods that accept LatLng objects also accept them in a simple Array form and simple object form (unless noted otherwise).

That may be true but for this specific case the function to be used is _explicitly_ called `latLngsToCoords`. It does not make sense to me to use an array when the function is named after this specific type and also used for converting a `LatLng` to an array of coordinates. Furthermore i tested your suggestion and **it does not work, the tests fail**. The code for `latLngToCoords` and `latLngsToCoords` uses properties of the latLng object (`lat`, `lng` and maybe `alt`) which obviously do not exist when you instead pass an array with just numbers.

> It's not so important (for my taste), because such type of error hard to make by accident, and it will manifest itself in some other part anyway.
> But it also makes sense to keep all your tests consistent, so...

Ok i can also change that, but i did not add this type check here because an array is not kind of a self invented object type like `LatLng`. But if you prefer i can add another `forEach` loop to these two lines and check for the array type...

Reply:
> It does not make sense to me to use an array when the function is named after this specific type

There may be some use cases.

> Furthermore i tested your suggestion and **it does not work, the tests fail**.

Good to know, perhaps this should fall into "unless noted otherwise", so we need to update docs.

> and check for the array type...

No need to check for Array, but you may want to check that it contains proper LatLngs.

Reply:
> No need to check for Array, but you may want to check that it contains proper LatLngs.

Sorry mate, i don't get it... What is the point of this? Have you looked at the [respective lines in the source code](https://github.com/Leaflet/Leaflet/blob/436430db4203a350601e002c8de6a41fae15a4bf/src/layer/GeoJSON.js#L256)? The function `latLngToCoords` or `latLngsToCoords` returns an array of coordinates/numbers not `LatLng`s. In the tests you mention i already check if the returned array contains the correct numbers that were given to the `LatLng` factory function. Where should i test then for proper `LatLng`s

Reply:
> The function `latLngToCoords` or `latLngsToCoords` returns an array of coordinates/numbers not `LatLng`s.

Again:
https://github.com/Leaflet/Leaflet/blob/cc74c2c8a11dcc4574afa1d746cb2dc0696a31ea/spec/suites/layer/GeoJSONSpec.js#L729

https://github.com/Leaflet/Leaflet/blob/cc74c2c8a11dcc4574afa1d746cb2dc0696a31ea/spec/suites/layer/GeoJSONSpec.js#L739

But you are right, no need to check types here, because they depend on `customCoordsToLatLng`.
So no objections left.

Reply:
> Good to know, perhaps this should fall into "unless noted otherwise", so we need to update docs.

#7425